### PR TITLE
Mockable status codes

### PR DESCRIPTION
**Pull request by LeopoldoFu**

### CODE REVIEW

**Comment by LeopoldoFu** [comment]
@ebsco/eac-software-engineers , I can't seem to find Evan on the list to assign this to him...

**Comment by LeopoldoFu** [comment]
Major, minor, patch version bump when this is merged? I'm thinking patch.

File: test.js [hunk 1]
```diff
@@ -58,8 +57,7 @@ describe('superagent mock', function() {
           data.should.have.property('id', '7');
           data.should.have.property('content', 'hello world, bitch!11');
```
Comment (pivnickm) [inline test.js]: Maybe we should edit this text for the EBSCO fork...

Comment (LeopoldoFu) [inline test.js]: Right...

**Comment by leachryanb** [comment]
What's this for?

**Comment by LeopoldoFu** [comment]
This a mocking library for unit testing the ajax library, [superagent](https://github.com/visionmedia/superagent), we're using in the react-redux POC.

**Comment by esimons** [comment]
Have you looked at [superagent-mock](https://github.com/M6Web/superagent-mock)? From the docs it looks like it supports sending status codes.

**Comment by LeopoldoFu** [comment]
I took a look at that (didn't try it), but wasn't sure if we could dynamically create mocks.  I like having the mocks be in-line with the tests (usually in beforeEach) instead of in a separate config file.

**Comment by esimons** [comment]
Looks like you should be able to configure mocks in-line with tests; the example in the README has the config object defined in a separate file, but it's still just being passed as an argument so it shouldn't matter.

**Comment by LeopoldoFu** [comment]
Well, I suppose I can spend some more time re-writing my tests and getting this to work...

**Comment by esimons** [comment]
Ah, hadn't realized you'd already committed to this library. Was just suggesting alternative in case it saved us having to fork something. Digging into it now.

**Comment by LeopoldoFu** [comment]
Well, I'm already (mentally) committed at looking at the other library now.

**Comment by esimons** [comment]
In either case, I'm sure the lib author would appreciate your work here.

**Comment by LeopoldoFu** [comment]
Switched all my tests over to superagent-mock.

**Comment by reggersusa** [comment]
Hi Leo,

These messages are coming to my personal email. Let me know how i can register only my work address.

Bob Eggers

**Comment by reggersusa** [comment]
Yeah, and why is this going to my personal email address?

Bob Eggers

**Comment by LeopoldoFu** [comment]
I believe it is linked to the email for your github account. Not sure if there’s anything to be done except change the email on your github account to your work email.

**Comment by reggersusa** [comment]
Thanks, will do.

Bob Eggers

**Comment by wylie** [comment]
@reggersusa you can have multiple email address associated with your GitHub account. https://github.com/settings/emails

I believe that if you are registered with the GitHub EBSCO organization under your @ebsco.com email address you will then get these notification emails there rather than your personal email.

**Comment by reggersusa** [comment]
Thanks! Not sure if I will. I never got this much email before. I feel loved!

Bob

**Pull request merged**